sdk/src/main/java/io/opentelemetry/sdk/trace/RandomIdsGenerator.java

```diff
@@ -41,7 +39,7 @@ public TraceId generateTraceId() {
     do {
       idHi = random.nextLong();
       idLo = random.nextLong();
-    } while (idHi == INVALID_ID && idLo == INVALID_ID);
+    } while ((idHi | idLo) == INVALID_ID);
```

Comment:
i'm curious...would you mind writing a little microbenchmark for this to see which is faster?

Reply:
This is a or not a xor, and is also correct compared to xor, so please update the description

Reply:
I shouldn't be coding that late :) fixed.

Reply:
@jkwatson, the microbenchmarks show that the legacy performed better, and it's also more readable, so I rolled it back.

Reply:
aha! I suspected as much. :)